api/v0/ingest/schema/envelope.go

```diff
@@ -150,5 +290,37 @@ func (ad *Advertisement) VerifySignature() (peer.ID, error) {
 		log.Warnw("advertisement has deprecated signature format", "signer", signerID)
 	}
 
+	if ad.ExtendedProvider != nil {
+		rec := &epSignatureRecord{}
+		// The top level provider must appear in the list of extended providers otherwise the ad is considered invalid
+		seenTopLevelProv := false
+		for _, p := range ad.ExtendedProvider.Providers {
```

Comment:
Do we need to check for duplicate providers here?

Reply:
Good point. I will add a logic to deduplicate on the ingestion side. Duplicates aren't prohibited by the spec itself it's more of an optimisation.